hypervisor/arch/x86/configs/apl-mrb/ve820.c

```diff
@@ -7,33 +7,33 @@
 #include <e820.h>
 
 const struct e820_entry ve820_entry[E820_MAX_ENTRIES] = {
-	{	/* 0 to mptable */
+	{	/* usable RAM under 1MB */
 		.baseaddr = 0x0UL,
-		.length   = 0xEFFFFUL,
+		.length   = 0xF0000UL,		/* 960KB */
 		.type     = E820_TYPE_RAM
 	},
 
-	{	/* mptable 65536U */
-		.baseaddr = 0xF0000UL,
-		.length   = 0x10000UL,
+	{	/* mptable */
+		.baseaddr = 0xF0000UL,		/* 960KB */
+		.length   = 0x10000UL,		/* 16KB */
 		.type     = E820_TYPE_RESERVED
 	},
 
-	{	/* mptable to lowmem */
-		.baseaddr = 0x100000UL,
-		.length   = 0x1FF00000UL,
+	{	/* lowmem */
+		.baseaddr = 0x200000UL,		/* 2MB */
+		.length   = 0x1FF00000UL,	/* 511MB */
 		.type     = E820_TYPE_RAM
 	},
 
-	{	/* lowmem to PCI hole */
-		.baseaddr = 0x20000000UL,
-		.length   = 0xa0000000UL,
+	{	/* between lowmem and PCI hole */
+		.baseaddr = 0x20100000UL,	/* 513MB */
+		.length   = 0x9FF00000UL,	/* 2559GB */
 		.type     = E820_TYPE_RESERVED
 	},
 
-	{	/* PCI hole to 4G */
-		.baseaddr = 0xe0000000UL,
-		.length   = 0x20000000UL,
+	{	/* between PCI hole and 4GB */
+		.baseaddr = 0xe0000000UL,	/* 3.5GB */
+		.length   = 0x20000000UL,	/* 512MB */
```

Comment:
This is good

Reply:
BTW, should we use (baseaddr, length), or (start_address, end_address)?

Reply:
There are two reasons for (base, size, type): 1) The Linux zero page structure is defined in this way so we can easily copy our e820 to zero page. 2) "Address Range Descriptor" in INT 15h E820h has same content.